openpgp-dsm/src/lib.rs

```diff
@@ -672,27 +684,47 @@ pub fn generate_key(
     )
     .context("could not create primary key")?;
 
-    info!("key generation: create signing subkey");
-    let signing_subkey = PublicKey::create(
-        &dsm_client,
-        key_name.to_string(),
-        KeyRole::SigningSubkey,
-        &algorithm,
-        exportable,
-        validity_period
-    )?;
+    let mut signing_subkeys: Vec<PublicKey> = vec![];
+    let mut encryption_subkeys: Vec<PublicKey> = vec![];
+    for flag in &key_flag_args {
+        match flag {
+            KeyFlagsArg::S => {
+                info!("key generation: create signing subkey");
+                let signing_subkey = PublicKey::create(
+                    &dsm_client,
+                    key_name.to_string(),
+                    KeyRole::SigningSubkey,
+                    &algorithm,
+                    exportable,
+                    validity_period
+                )?;
 
-    info!("key generation: create decryption subkey");
-    let encryption_subkey = PublicKey::create(
-        &dsm_client,
-        key_name.to_string(),
-        KeyRole::EncryptionSubkey,
-        &algorithm,
-        exportable,
-        validity_period
-    )?;
+                signing_subkeys.push(signing_subkey);
+            },
 
-    let prim_flags = KeyFlags::empty().set_certification();
+            KeyFlagsArg::EtEr => {
+                info!("key generation: create decryption subkey");
+                let encryption_subkey = PublicKey::create(
+                    &dsm_client,
+                    key_name.to_string(),
+                    KeyRole::EncryptionSubkey,
+                    &algorithm,
+                    exportable,
+                    validity_period
+                )?;
+
+                encryption_subkeys.push(encryption_subkey);
+            },
+
+            _ => {},
```

Comment:
Do we need to consider other subkey types? Like C or CS? (Or are those not valid subkeys?)

(Sorry, I'm not an expert here)

Reply:
ok, please ignore my previous comment. subkeys shouldn't be used for certification :man_facepalming: